### DISCUSSION TITLE
Add jump command with range limited to the current cursor line (easymotion-fl)

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#10534] [General]
What I do not like about new jump command (```gw```) is that
1) the word search is not limited to the region/line where I'm typing
2) it has no directionality ("Am I searching either forward or backward??")
3) you can't jump on any char but only words' beginning
4) I have to type at least 4 chars to move the cursor....

What I would like Helix to have is an overloaded ```f``` like in the following example:
Suppose I am with cursor on the first ```i``` of this line
```int main()```
If I typed in normal mode ```fa```, then the cursor would move onto the ```a```.

Suppose I am with cursor on the first ```i``` of this line
```int main(int argc, char **argv)```
If I typed in normal mode ```fa```, then all the ```a``` would be temporarily turned into other chars, and these ones would be highlighted,
like here (let's pretend capital case is highlighting )
```int mBin(int Crgc, chDr **Ergv)```
then I have to type one of the chars B, C, D or E to move the cursor where I want onto. Obviously, after that everything is restored.

Suppose I am with cursor on the ```m``` of this line
```int main(int argc, char **argv)```
If I typed in normal mode ```fi```, then the cursor would move onto the ```i``` of the second```int```.

I just described ```easymotion-fl``` behavior, which I really miss ;(
It would definitely be a game changer
https://github.com/easymotion/vim-easymotion/blob/master/doc/easymotion.txt#L496

**Comment by pascalkuthe** [comment]
The jump command in core is opiniates and purposefully simple. We will not add a bunch of alternative jump commands to core that is not maintainable

**Reply by ghost** [reply]
I invite you to re-consider my idea. You can't really type 4 characters to jump (and you can't jump everywhere neither!!). I repeat myself: It would be a game changer IMHO

**Reply by ghost** [reply]
with ```gw``` I can't jump on single char words.
e.g. here ```perros y gatos``` (dogs and cats in spanish) I am not able to jump onto ```y```!!!...
